src/main/kotlin/net/atos/zac/app/zaken/converter/historie/RESTZaakHistoriePartialUpdateConverter.kt

```diff
@@ -5,42 +5,48 @@ import net.atos.client.vrl.VrlClientService
 import net.atos.client.vrl.model.generated.CommunicatieKanaal
 import net.atos.client.zgw.shared.model.audit.ZRCAuditTrailRegel
 import net.atos.client.zgw.shared.util.URIUtil
+import net.atos.client.zgw.zrc.ZRCClientService
 import net.atos.client.zgw.zrc.model.Geometry
 import net.atos.zac.app.audit.model.RESTHistorieActie
 import net.atos.zac.app.audit.model.RESTHistorieRegel
 import java.net.URI
 
 private const val COMMUNICATIEKANAAL = "communicatiekanaal"
 private const val ZAAKGEOMETRIE = "zaakgeometrie"
+private const val HOOFDZAAK = "hoofdzaak"
+private const val RELEVANTE_ANDERE_ZAKEN = "relevanteAndereZaken"
 
 class RESTZaakHistoriePartialUpdateConverter @Inject constructor(
-    private val vrlClientService: VrlClientService
+    private val vrlClientService: VrlClientService,
+    private val zrcClientService: ZRCClientService
 ) {
     fun convertPartialUpdate(
         auditTrail: ZRCAuditTrailRegel,
         actie: RESTHistorieActie?,
         old: Map<*, *>,
         new: Map<*, *>
     ) =
-        old.diff(new).map { convertLine(auditTrail, actie, it) }
+        old.diff(new).mapNotNull { convertLine(auditTrail, actie, it) }
 
     private fun convertLine(
         auditTrail: ZRCAuditTrailRegel,
         actie: RESTHistorieActie?,
-        change: Map.Entry<Any?, Pair<*, Any?>>
```

Comment:
how can you have a change entry without a key? is that not a bit weird?

Reply:
It will never be null but Kotlin complains if I change it to just Any, because earlier in the call chain I can't cast to a map with a non-nullable key

Reply:
This applies to your other comments as well. I propose we look at improving that in another PR.